gui/wxpython/mapdisp/frame.py

```diff
@@ -659,6 +666,21 @@ def RemoveQueryLayer(self):
         for layer in qlayer:
             self.GetMap().DeleteLayer(layer)
 
+    def SetDockingCallback(self, function):
+        """Sets docking bound method to dock or undock"""
+        self.docking_callback = function
+
+    def OnDockUndock(self, event):
+        """Dock or undock map display panel to independent MapFrame"""
+        self.docking_callback(self)
```

Comment:
It looks like the docking_callback may (theoretically) not be defined when this method is called, so I suggest to initialize it to None and test here before calling it.